Components Showcase page revamp (#6271)
hicommonwealth/commonwealth: knowledge_base/Component-Kit.md

A rendered version of our component kit can be [viewed on the live site](https://commonwealth.im/components), or found in the codebase in the `component_kit` directory.

> Components page got revamped and this also means that we will structure the components in a bit more readable way. There are two groups of building blocks in our kit:

Reviewer: Possible to add a specific datestamp here? E.g. "As of 240109, the components page has been revamped"

Author: done

- [Foundations](https://www.figma.com/file/yCUKk46hg9Fy3EXGkSoaKT/%F0%9F%9A%A7-Foundation?type=design&node-id=2%3A27&mode=design&t=qnBiTOOKaYhIM9bI-1) - the visual elements needed to create engaging end-to-end user experiences. This includes guidance on iconography, typography, layout and structure.
- [Components](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&mode=design&t=LQy3p5zpdAM5fos8-1) - reusable building blocks of our design system. Each component meets a specific interaction or UI need, and has been specifically created to work together to create patterns and intuitive user experiences.

### How to add component to components showcase page?

> 1. Being in `/commonwealth` package, run `yarn add-component-showcase` command

Reviewer: ```suggestion
1. From within the `/commonwealth` package, run the `yarn add-component-showcase` command
```

Author: done

2. Type a name for the new component. Make sure it is aligned with the design team. This name will be used as a file name in the codebase and a display name in component showcase page.
3. Pick if new component should be of type "Foundations" or "Components". This should be aligned with the design team.
4. New file is created in `/ComponentsShowcase/components` directory and `/ComponentsShowcase/componentsList.ts` file is updated.
5. Edit newly created file to add content (display new component).

### Engineering tips
- In generated `{Component}.showcase.tsx` file use React Fragment `<> ... </>` as a wrapper for the whole component. 

> - Each generated file comes along with generated CSS class name that is identical as file name. If you want to add styles to specific showcase component, add proper class and styling in `ComponentsSwhocase.scss`.

Reviewer: ```suggestion
- Each generated file comes along with a generated CSS class name that is identical to the file name. If you want to add styles to specific showcase component, add proper class and styling in `ComponentsSwhocase.scss`.
```

- There are two main util classes for handling the layout of this page (`flex-column` and `flex-row`). The idea is to minimize customization of styles in this page. Everything should be auto or semi generated.   

> ## Change Log

Reviewer: ```suggestion

## Change Log
```

Author: done

- 240109: Authored by Marcin Maslanka.

> - 230823: Authored by Graham Johnson.

Reviewer: ```suggestion
```

Feel free to remove me in the change log—I basically just set up the page stub and probably shouldn't have been listed as author to begin with.

Author: done
